### PR TITLE
BUG: Clean app state after crash (Mosviz)

### LINKED ISSUE #364: loading correct data after exception does not properly load in mosviz
Loading the correct data after an previous attempt that throws an exception does not properly load the data.  I would expect all data containers to cleanse and purge if an exception was thrown during initial data loading.   Instead I need to reinitialize the application.  In the example below it generates the same traceback as indicated in #363. 

```
m = MosViz()
m.app

# initial incorrect data loading
m.load_data(spectra_1d=w1d, spectra_2d=w2d, images=[])

# get a traceback error

# correct data loading
m.load_data(spectra_1d=w1d, spectra_2d=w2d, images=cutouts)

# throws same traceback error
```

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to clear Application's dirty state when data loading fails.

The original fix was at #830 and only for Mosviz but @havok2063 said it might be beneficial to just do this for all the viz.

Fixes #364

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/app.py [hunk 1]
```diff
@@ -327,6 +327,9 @@ def load_data(self, file_obj, parser_reference=None, **kwargs):
             snackbar_message = SnackbarMessage("Data successfully loaded.",
                                                sender=self)
             self.hub.broadcast(snackbar_message)
+        except Exception:  # Reset state on uncaught errors
+            self.data_collection.clear()
```
Comment (javerbukh) [inline jdaviz/app.py:331]: So if I load data 2 separate times and this fails on the second time, will ALL data be removed from data_collection?

Comment (pllim) [reply jdaviz/app.py:331]: Good point... I'll have to ponder a bit more.

Comment (pllim) [reply jdaviz/app.py]: I think I addressed it in second commit. Thanks for pointing that out!

File: jdaviz/app.py [hunk 2]
```diff
@@ -327,6 +327,11 @@ def load_data(self, file_obj, parser_reference=None, **kwargs):
             snackbar_message = SnackbarMessage("Data successfully loaded.",
                                                sender=self)
             self.hub.broadcast(snackbar_message)
+        except Exception:  # Reset state on uncaught errors
+            cfg_name = self.state.settings.get('configuration', 'unknown')
+            if cfg_name in ('mosviz', ):  # Add more as needed.
```
Comment (pllim) [inline jdaviz/app.py:332]: This keeps the fix in upper-level app code but allow us to pick and choose which viz get that fix.

**Review by havok2063 — APPROVED** [review]
Looks good to me.

**Review by javerbukh — APPROVED** [review]
Works (or doesn't, in this case) as expected!

**Pull request merged**